client/consensus/nimbus-consensus/src/lib.rs

```diff
@@ -76,24 +61,7 @@ where
 	Client: ProvideRuntimeApi<Block>,
 	Client::Api: NimbusApi<Block>,
 {
-	// Determine if runtime change
-	let runtime_upgraded = if *parent.number() > sp_runtime::traits::Zero::zero() {
-		use sp_api::Core as _;
-		let previous_runtime_version: sp_version::RuntimeVersion = para_client
-			.runtime_api()
-			.version(parent.hash())
```

Comment:
Instead of removing the code, simply replace parent.hash() with parent.parent_hash(). This will fix the issue and reintroduce the hold behavior needed to prevent chain stalls when we change the author-eligibility algorithm.

Reply:
This was broken in the following commit: https://github.com/Moonsong-Labs/moonkit/commit/79daf0f64e92d2b6f625dd3cfaf0067bcf1ec701

I still would prefer to remove it, since having every collator forcing collation on runtime upgrades will cause unnecessary gossiping in the P2P network.

Addressed in bd168bf

Reply:
>  having every collator forcing collation on runtime upgrades will cause unnecessary gossiping in the P2P network.

Only collators that are actually eligible will gossip their block to the network, because if they are not eligible, the block builder task will fail. Therefore, this fix does not cause unnecessary gossiping.